packages/libs/react-ui/src/components/InputWrapper/InputWrapper.tsx

```diff
@@ -39,15 +39,17 @@ export const InputWrapper: FC<IInputWrapperProps> = ({
         <InputHeader htmlFor={htmlFor} label={label} tag={tag} info={info} />
       )}
       <div className="inputGroup">
-        {React.Children.map(children, (child) => {
-          if (!React.isValidElement(child)) return null;
-          const props = {
-            ...child.props,
-            leadingTextWidth,
-          };
+        {leadingTextWidth
+          ? React.Children.map(children, (child) => {
+              if (!React.isValidElement(child)) return null;
+              const props = {
+                ...child.props,
+                leadingTextWidth,
+              };
 
-          return React.cloneElement(child, props);
-        })}
+              return React.cloneElement(child, props);
```

Comment:
We're likely switching to using the Context API for passing props into subcomponents. We could refactor this later or, if you like, you can have a look at how I did it here: https://github.com/kadena-community/kadena.js/pull/954

Reply:
Here is the ticket for it: https://app.asana.com/0/1205693948735844/1205693948735848